README.md

```diff
@@ -1,38 +1,147 @@
+    * Corrupt datasets can be noted by the warning log `Error: Response not of correct format: Response does not have rows field`.
+    * This can be reflected on the **Network** tab on F12, in which the exact reason can be spotted.
+    * It is possible for a .db to work normally on My Datasets, but regarded as corrupt on Visualize.
+    * An example is `crimedata_corruptisstate.db`, in which the requets fails due to lacking the `isState` flag only used in the Visualization tab.
+    * All non-db files are removed upon performing upload.
+* It is currently a hassle to manually input all the data into a CSV.
+
+# My Datasets
+The user is able to view all of the data after selecting or reselecting the databases processed and stored on the backend. The most important data are shown on the table, further data can be seen by hovering over the community name.
+
+The displayed data is derived from the response after calling `http://localhost:3000/mapData?allLevel&datasetID=$DBNAME` where `$DBNAME` is the name of the database file as found in `$BACKEND/db/`. The response will be a collection of **rows**, where each row is a JSON object containing attributes corresponding to the column names found in the `locations` table of that `.db` file.
+
+If the `.db` file is not of the correct format, such as lacking the `locations` table, or being a folder, a 418 TEAPOT response will be sent by the backend.
+
+![Upload](notes/Screenshots/AppMyDatasets.png)
+
+## Temporarily Restrictions
+* The UX experience for viewing all the data is currently suboptimal
+    * With 150 fields, the tooltip may overflow
+* Sorting and showing/hiding column are currently not supported
+* The entire table will be loaded, making the experience slow for large datasets
+* The fetch will always fail and return a HTTP 300 if the `db` file contains only whitespaces, or is named `"null"`.
+* All files on the `db` folder will be listed. Currently, the backend glitches, sends out 500, but refuses to process anyways when a folder is used.
+
+# Visualize
+The capstone of our project. The user is provided a map, implemented in React Leaflet. After selecting a file, the app will load the polygons to provide a crime frequency choropleth. A dropdown is provided at the top to filter the visualizations; a backend fetch is done each time you change the dataset.
+
+The map visualization is implemented in **React Leaflet**, the **React** implementation of a popular open-source JS mapping library. The `<Map>` component will take in the name of the dataset as a prop, and consequently perform the fetch, to load the (multi)polygons for each city. It then color-codes each polygon depending on the calculated overall crime rate. You can click on each polygon to view the exact population and crime statistics, including the overall number of violent crimes committed in this region.
+
+## Current coloring scheme
```

Comment:
We said we wouldn't do this, remove 114-123

Reply:
Removed. We just said "a formula"